vortex-scalar/src/pvalue.rs

```diff
@@ -21,6 +22,20 @@ pub enum PValue {
     F64(f64),
 }
 
+macro_rules! as_primitive {
+    ($T:ty, $PT:tt) => {
+        paste! {
+            pub fn [<as_ $T>](self) -> Option<$T> {
```

Comment:
```suggestion
            #[doc = "Access PValue as `" $T "`, returning `None` if conversion is unsuccessful"]
            pub fn [<as_ $T>](self) -> Option<$T> {
```

Reply:
Ooo, I wasn't sure how to do this. Will make a pr

Reply:
Made https://github.com/spiraldb/vortex/pull/658 since this pr merged